File: Sample Algorithms/Exam Study/Recursion Exercies/3.c
// Write a recursive program that inverts the order of the elements in the array

#include <stdio.h>

void inverseArray(char arr[],int start,int end){
    if (start == end || start > end){
        return ;
    }
    printf("Switching: %c with %c\n",arr[start],arr[end]);
    char temp = arr[start];
    arr[start] = arr[end];
    arr[end] = temp;
    inverseArray(arr,start+1,end-1);
}

void main () {
    char array[] = "domo arigatou mr.robato";
    int arrSize = (sizeof(array)/sizeof(array[0]) -2);
    inverseArray(array,0,arrSize);
    printf("Inversed Array: %s",array);
}

File: Sample Algorithms/Exam Study/Recursion Exercies/3.py
# Write a recursive program that inverts the order of the elements in the array


a = ['d','o','m','o','a','r','r']
b = lambda a: (b (a[1:]) + a[:1] if a else[])

print (b(a))


def reverse(a):
    if len(a) == 0:
        return []
    return [a[-1]] + reverse(a[:-1])

    
a = [1,2,3,4,5]

print(reverse(a))